### PR TITLE
Stop exposing fetch types in HttpOperationResponse

### PR DESCRIPTION
**Pull request by RikkiGibson**
Based on Azure/ms-rest-js#84

Also:

- adds a single "unified" testserver which serves as a proxy to autorest.testserver as well as webpack-dev-middleware
- Reenables files tests, splitting them into browser-specific and nodejs-specific variants.

### CODE REVIEW

File: test/vanilla/acceptanceTests.ts [hunk 1]
```diff
@@ -1763,75 +1764,165 @@ describe('typescript', function () {
     describe('Files Client', function () {
       var testClient = new AutoRestSwaggerBATFileService(baseUri, clientOptions);
 
-      it('should correctly deserialize binary streams', function (done) {
```
Comment (RikkiGibson) [inline test/vanilla/acceptanceTests.ts]: these changes are worth a look

File: webpack.config.ts [hunk 2]
```diff
@@ -11,7 +12,7 @@ const config: webpack.Configuration = {
     path: __dirname
   },
   plugins: [
-    new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/)
+    new ForkTsCheckerWebpackPlugin({ silent: true })
```
Comment (RikkiGibson) [inline webpack.config.ts]: This makes it so the webpack bundle doesn't wait on typescript type checking. Brings my test bundle time from about 22s -> 15s. Errors get reported in the testserver's stdout based on the forked checker process's output.

File: webpack.config.ts [hunk 3]
```diff
@@ -38,7 +43,7 @@ const config: webpack.Configuration = {
     }
   },
   node: {
-    fs: false,
+    fs: "empty",
```
Comment (RikkiGibson) [inline webpack.config.ts]: This lets us just ignore fs in the browser environment--as long as we don't actually run a line of code which uses fs, we're all good.

File: package.json [hunk 4]
```diff
@@ -53,8 +58,8 @@
     "mocha": "^5.0.4",
     "mocha-chrome": "^1.0.3",
     "moment": "~2.21.0",
-    "ms-rest-azure-js": "Azure/ms-rest-azure-js#GitHubReference",
-    "ms-rest-js": "Azure/ms-rest-js",
+    "ms-rest-azure-js": "RikkiGibson/ms-rest-azure-js#HttpClient",
+    "ms-rest-js": "RikkiGibson/ms-rest-js#HttpClient",
```
Comment (daschult) [inline package.json]: Until @amarzavery approves publishing a new version of ms-rest-js, we really should leave these pointing to ms-rest-js and ms-rest-azure-js#GitHubReference.

File: test/vanilla/acceptanceTests.ts [hunk 5]
```diff
@@ -1763,75 +1764,165 @@ describe('typescript', function () {
...
+          const actualBytes = new Uint8Array(reader.result);
+          const expectedBytes = new Uint8Array(content);
+          actualBytes.length.should.equal(expectedBytes.length, 'length');
+          for (let i = 0; i < actualBytes.length; i++) {
+            actualBytes[i].should.equal(expectedBytes[i], `position ${i}`);
+          }
+        });
+      }
+
+      if (msRest.isNode) {
```
Comment (daschult) [inline test/vanilla/acceptanceTests.ts]: Would it make sense to collect all of the node tests into a single block?

File: package.json (hunk 4 above)
Comment (RikkiGibson) [reply package.json]: This is to make CI pass so we can confirm these changes are ok before merging/publishing on the ms-rest-js side.

**Comment by azuresdkci1x** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.5.11*

## ❌️ General: 90%

<details><summary>53 out of 535 features are not covered by tests</summary><p>

[`HttpClientFailure400Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Delete), [`HttpClientFailure400Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Get), [`HttpClientFailure400Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Head), [`HttpClientFailure400Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Patch), [`HttpClientFailure400Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Post), [`HttpClientFailure400Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Put), [`HttpClientFailure401Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure401Head), [`HttpClientFailure402Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure402Get), [`HttpClientFailure403Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure403Get), [`HttpClientFailure404Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure404Put), [`HttpClientFailure405Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure405Patch), [`HttpClientFailure406Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure406Post), [`HttpClientFailure407Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure407Delete), [`HttpClientFailure409Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure409Put), [`HttpClientFailure410Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure410Head), [`HttpClientFailure411Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure411Get), [`HttpClientFailure412Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure412Get), [`HttpClientFailure413Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure413Put), [`HttpClientFailure414Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure414Patch), [`HttpClientFailure415Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure415Post), [`HttpClientFailure416Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure416Get), [`HttpClientFailure417Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure417Delete), [`HttpClientFailure429Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure429Head), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`ResponsesScenarioEmptyErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioEmptyErrorBody), [`ResponsesScenarioNoModelErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioNoModelErrorBody), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`additionalPropertiesInProperties`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInProperties), [`additionalPropertiesInPropertiesWithAPTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInPropertiesWithAPTypeString), [`additionalPropertiesTrue`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTrue), [`additionalPropertiesTypeObject`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeObject), [`additionalPropertiesTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeString), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`animalNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=animalNotFoundError), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`expectedNoErrors`](https://github.com/Azure/autorest.testserver/search?q=expectedNoErrors), [`expectedPetHungryError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetHungryError), [`expectedPetSadError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetSadError), [`getDurationInvalid`](https://github.com/Azure/autorest.testserver/search?q=getDurationInvalid), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`intError`](https://github.com/Azure/autorest.testserver/search?q=intError), [`linkNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=linkNotFoundError), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putComplexPolymorphismNoDiscriminator`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismNoDiscriminator), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`stringError`](https://github.com/Azure/autorest.testserver/search?q=stringError), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 99%

<details><summary>1 out of 118 features are not covered by tests</summary><p>

[`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Comment by RikkiGibson** [comment]
Linux build is really unhappy.. stuff like this in there:

```sh
2018-05-15T20:49:16.9895300Z test/azure/generated/AzureCompositeModelClient/azureCompositeModel.ts(13,25): error TS2307: Cannot find module 'ms-rest-js'.
2018-05-15T20:49:16.9993681Z test/azure/generated/AzureCompositeModelClient/azureCompositeModel.ts(14,30): error TS2307: Cannot find module 'ms-rest-azure-js'.
```

Will try the CI build again after ms-rest-js and ms-rest-azure-js are merged.

**Comment by daschult** [comment]
I’ve been seeing that recently. I have no idea what is wrong with that Linux box. The windows box works fine.

Dan Schulte

**Comment by RikkiGibson** [comment]
Any concerns still @daschult ? (maybe refactoring related to grouping browser and node tests together, vs grouping equivalent feature tests together, can maybe wait until a little further down the line?)

**Comment by daschult** [comment]
@RikkiGibson Nope, the package.json was my biggest concern. Thanks for taking care of that.

**Comment by azuresdkci1x** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.5.11*

## ❌️ General: 90%

<details><summary>53 out of 535 features are not covered by tests</summary><p>

[`HttpClientFailure400Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Delete), [`HttpClientFailure400Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Get), [`HttpClientFailure400Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Head), [`HttpClientFailure400Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Patch), [`HttpClientFailure400Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Post), [`HttpClientFailure400Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure400Put), [`HttpClientFailure401Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure401Head), [`HttpClientFailure402Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure402Get), [`HttpClientFailure403Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure403Get), [`HttpClientFailure404Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure404Put), [`HttpClientFailure405Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure405Patch), [`HttpClientFailure406Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure406Post), [`HttpClientFailure407Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure407Delete), [`HttpClientFailure409Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure409Put), [`HttpClientFailure410Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure410Head), [`HttpClientFailure411Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure411Get), [`HttpClientFailure412Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure412Get), [`HttpClientFailure413Put`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure413Put), [`HttpClientFailure414Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure414Patch), [`HttpClientFailure415Post`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure415Post), [`HttpClientFailure416Get`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure416Get), [`HttpClientFailure417Delete`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure417Delete), [`HttpClientFailure429Head`](https://github.com/Azure/autorest.testserver/search?q=HttpClientFailure429Head), [`HttpRedirect301Put`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect301Put), [`HttpRedirect302Patch`](https://github.com/Azure/autorest.testserver/search?q=HttpRedirect302Patch), [`ResponsesScenarioEmptyErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioEmptyErrorBody), [`ResponsesScenarioNoModelErrorBody`](https://github.com/Azure/autorest.testserver/search?q=ResponsesScenarioNoModelErrorBody), [`UrlQueriesArrayMultiEmpty`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiEmpty), [`UrlQueriesArrayMultiNull`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiNull), [`UrlQueriesArrayMultiValid`](https://github.com/Azure/autorest.testserver/search?q=UrlQueriesArrayMultiValid), [`additionalPropertiesInProperties`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInProperties), [`additionalPropertiesInPropertiesWithAPTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesInPropertiesWithAPTypeString), [`additionalPropertiesTrue`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTrue), [`additionalPropertiesTypeObject`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeObject), [`additionalPropertiesTypeString`](https://github.com/Azure/autorest.testserver/search?q=additionalPropertiesTypeString), [`allowedValueEnum`](https://github.com/Azure/autorest.testserver/search?q=allowedValueEnum), [`animalNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=animalNotFoundError), [`expectedEnum`](https://github.com/Azure/autorest.testserver/search?q=expectedEnum), [`expectedNoErrors`](https://github.com/Azure/autorest.testserver/search?q=expectedNoErrors), [`expectedPetHungryError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetHungryError), [`expectedPetSadError`](https://github.com/Azure/autorest.testserver/search?q=expectedPetSadError), [`getDurationInvalid`](https://github.com/Azure/autorest.testserver/search?q=getDurationInvalid), [`getEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferenced), [`getEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=getEnumReferencedConstant), [`intError`](https://github.com/Azure/autorest.testserver/search?q=intError), [`linkNotFoundError`](https://github.com/Azure/autorest.testserver/search?q=linkNotFoundError), [`putComplexPolymorphismComplicated`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismComplicated), [`putComplexPolymorphismNoDiscriminator`](https://github.com/Azure/autorest.testserver/search?q=putComplexPolymorphismNoDiscriminator), [`putEnumReferenced`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferenced), [`putEnumReferencedConstant`](https://github.com/Azure/autorest.testserver/search?q=putEnumReferencedConstant), [`roundTripEnum`](https://github.com/Azure/autorest.testserver/search?q=roundTripEnum), [`stringError`](https://github.com/Azure/autorest.testserver/search?q=stringError), [`unexpectedEnum`](https://github.com/Azure/autorest.testserver/search?q=unexpectedEnum)</p></details>

## ❌️ Azure: 99%

<details><summary>1 out of 118 features are not covered by tests</summary><p>

[`LRONonRetryPut201Creating400InvalidJson`](https://github.com/Azure/autorest.testserver/search?q=LRONonRetryPut201Creating400InvalidJson)</p></details>

**Pull request merged**